packages/application-generic/src/usecases/add-job/add-job.usecase.ts

```diff
@@ -54,22 +62,47 @@ export class AddJob {
 
     Logger.log(`Starting New Job ${job._id} of type: ${job.type}`, LOG_CONTEXT);
 
-    const digestAmount =
-      job.type === StepTypeEnum.DIGEST
-        ? await this.addDigestJob.execute(AddDigestJobCommand.create({ job }))
-        : undefined;
-
+    let digestAmount: number | undefined;
+    let digestCreationResult: DigestCreationResultEnum | undefined;
     if (job.type === StepTypeEnum.DIGEST) {
+      validateDigest(job);
+      digestAmount = this.calculateDelayService.calculateDelay({
```

Comment:
I consider this should live inside `MergeOrCreateDigestUsecase` and the output of that use case to be an object with:
- digestAmount
- digestCreationResult

Reply:
I think MergeOrCreateDigestUsecase should only return the status for the step as it does now because it is now very easy to follow on a high level of what we are doing with a digest step.

Reply:
Yeah, that was also our consideration during the implementation. But it felt that the function was doing 2 distinguished actions. One, is to `MergeOrCreateDigestUsecase` which is as expected, and the second is to calculate the time. 

I can see a reason to return an object instead. Giving it a second thought, it might be better to return object and incapsulate the logic of the creation of digest to be responsible for the "How long" we should delay it too. 

@davidsoderberg I think the previous issue was that the response type was very confusing. And we acted on "null" or number for multiple states. Having an object return and an optional key for the digestAmount will handle it nicely.

wdyt?

Reply:
I still think there is a reason to keep them separate to make it easy to understand what happens on add job